src/main/java/io/confluent/connect/jdbc/source/TableMonitorThread.java

```diff
@@ -150,6 +162,24 @@ private synchronized boolean updateTables() {
       filteredTables.addAll(tables);
     }
 
+    Map<String, List<TableId>> duplicates = filteredTables.stream()
+        .collect(Collectors.groupingBy(TableId::tableName))
+        .entrySet().stream()
+        .filter(entry -> entry.getValue().size() > 1)
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    if (!duplicates.isEmpty()) {
+      log.info("The filtered tables are {} ", filteredTables);
+      log.error("Connector uses unqualified table name as the topic name and has detected "
+          + "duplicate unqualified table names. This could lead to mixed data type and "
+          + "subsequently errors during the processing. Hence to prevent processing errors, "
+          + "JDBC Source connector fails fast fr such configurations. To be able to start the "
+          + "connector, update connector's {} config to include exactly one table in each of the "
+          + "below listed tables.\n {} ", configErrorMsg, duplicates);
+      // this will force the connector to fail configuration as it doesn't have any table to process
+      filteredTables.clear();
```

Comment:
That makes sense, but the actual failure message will be "Number of groups must be positive", which is not going to be intuitive. How about if we set a flag here (perhaps the duplicate tableIDs), and then throw this error message in the `tables()` method. Both `updateTables()` and `tables()` are synchronized, so there shouldn't be any atomicity or visibility problems.